webrtc/interfaces.html

```diff
@@ -0,0 +1,118 @@
+       }
+
+       function initRTCSessionDescription(idlArray) {
+         window.sessionDesc = new RTCSessionDescription({ type: 'offer' });
+         idlArray.add_objects({'RTCSessionDescription': ['sessionDesc']});
+       }
+
+       function initRTCIceEvent(idlArray) {
+         window.iceEvent = new RTCPeerConnectionIceEvent('ice-event', { });
+         idlArray.add_objects({'RTCPeerConnectionIceEvent': ['iceEvent']});
+       }
+
+       function initRTCIceErrorEvent(idlArray) {
+         window.iceErrorEvent = new RTCPeerConnectionIceErrorEvent('ice-error', { errorCode: 701 });
+         idlArray.add_objects({'RTCPeerConnectionIceErrorEvent': ['iceErrorEvent']});
+       }
+
+       function initRTCCertificate(idlArray) {
+         return RTCPeerConnection.generateCertificate({
+           name: 'RSASSA-PKCS1-v1_5',
+           modulusLength: 2048,
+           publicExponent: new Uint8Array([1, 0, 1]),
+           hash: 'SHA-256'
+         })
+         .then(cert => {
+           window.rtcCert = cert;
+           idlArray.add_objects({'RTCCertificate': ['rtcCert']});
+         });
+       }
+
+       const syncInits = [
+         ['Init RTCPeerConnection', initRTCPeerConnection],
+         ['Init RTCIceCandidate', initRTCIceCandidate],
+         ['Init RTCSessionDescription', initRTCSessionDescription],
+         ['Init RTCPeerConnectionIceEvent', initRTCIceEvent],
+         ['Init RTCPeerConnectionIceErrorEvent', initRTCIceErrorEvent],
+       ];
+
+       const asyncInits = [
+         ['Init RTCCertificate', initRTCCertificate],
+       ];
+
+       function initIdlArray(idlText) {
+         const idlArray = new IdlArray();
+
+         idlArray.add_untested_idls('interface EventHandler {};');
+         idlArray.add_untested_idls('interface MediaStreamTrack {};');
+         idlArray.add_idls(idlText);
+
+         return idlArray;
+       }
+
+       function doIdlTests(idlArray) {
+         for(const [taskName, initIdl] of syncInits) {
+           test(t => {
+             initIdl(idlArray);
+             t.done();
+           }, taskName);
+         }
+
+         // Async initialize all IDL instances in parallel
+         return Promise.all(
+           asyncInits.map(([taskName, initIdl]) => {
+             // TODO: fix promise_test so that it return a promise
+             // Otherwise there might be race condition when IDL test starts
+             return promise_test(
+               () => initIdl(idlArray),
+               taskName);
+           }))
+           .then(() => {
+             // Run the IDL array test regardless of whether
+             // any of the IDL  initialization failed
+             idlArray.test();
+           });
+       }
+
+       promise_test(() => {
+         return fetch('/interfaces/webrtc.idl')
+           .then(response => response.text())
+           .then(initIdlArray)
+           .then(doIdlTests);
+
+       }, 'WebRTC IDL Tests');
```

Comment:
Call this "Test driver" since it's just a mechanism to collect IDLs and set up the real tests. (Like https://github.com/ayg/web-platform-tests/blob/master/XMLHttpRequest/interfaces.html)